### PR TITLE
fix: autocomplete inserts duplicate prefix for dotted metric names

### LINKED ISSUE #493: Autocomplete inserts duplicate prefix for metric names containing dots
#### Problem

  When using autocomplete for metric names that contain dots (e.g. kubernetes.pod.id), selecting a suggestion produces a duplicated prefix.

  #### Steps to reproduce:

  1. Open the query editor
  2. Start typing a dotted metric name, e.g. kubernetes.pod
  3. Autocomplete suggests kubernetes.pod.id
  4. Select the suggestion

  Expected result: kubernetes.pod.id

  Actual result: kubernetes.kubernetes.pod.id

  #### Root cause

  Monaco's getWordAtPosition() relies on the wordPattern regex defined in promql.ts, which treats dots (.) as word separators. When the cursor is at kubernetes.pod|, Monaco identifies only pod as the current word.
  The completion replace range covers only pod, but insertText contains the full metric name kubernetes.pod.id. As a result, the text before the dot (kubernetes.) is preserved and the full metric name is inserted
  after it.

  #### Affected metrics

  Any metric name containing dots, commonly seen in environments using Kubernetes, OpenTelemetry, or other systems that use dot-separated naming conventions (e.g. container.cpu.usage, k8s.node.memory,
  process.runtime.gc.count).

### PR DESCRIPTION
**Pull request by arturminchukov**
Related issue: #493 

### Describe Your Changes

fix: autocomplete inserts duplicate prefix for dotted metric names

  Monaco's getWordAtPosition() treats dots as word boundaries, so for a
  metric like "kubernetes.pod.id" it only recognizes "pod" as the current
  word. The completion replace range covered only "pod", but insertText
  contained the full name, resulting in "kubernetes.kubernetes.pod.id".

  Fix by computing a custom range for metric name completions that scans
  backwards over dots to cover the entire dotted prefix the user has typed.

### Checklist

The following checks are **mandatory**:

- [x] My change adheres to [VictoriaMetrics contributing guidelines](https://docs.victoriametrics.com/victoriametrics/contributing/#pull-request-checklist).
- [x] My change adheres to [VictoriaMetrics development goals](https://docs.victoriametrics.com/victoriametrics/goals/).

---
## Summary by cubic
Fixes autocomplete for dotted and colon-separated metric names to prevent duplicate prefixes (e.g., avoids "kubernetes.kubernetes.pod.id"). Uses a custom replace range that spans the full identifier around the cursor.

- **Bug Fixes**
  - Replace the entire `[a-zA-Z0-9_.:]` identifier for `metricName` completions; other types keep Monaco’s default range.
  - Add unit tests for edge cases and update `CHANGELOG.md` (see #493).

<sup>Written for commit 856c81175713b305234f037a6fcbc62b35e48a58. Summary will update on new commits.</sup>

### CODE REVIEW

**Review by Loori-R — APPROVED** [review]

File: src/components/monaco-query-field/monaco-completion-provider/getDottedIdentifierBounds.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+// Find the start and end offsets (0-based) of the dotted identifier around the cursor.
+// Monaco's getWordAtPosition treats dots as word boundaries, so we scan manually.
+export function getDottedIdentifierBounds(lineContent: string, cursorCol0: number): { start: number; end: number } {
+  let start = cursorCol0;
+  while (start > 0 && /[a-zA-Z0-9_.:]/.test(lineContent[start - 1])) {
+    start--;
```
Comment (dmitryk-dk) [inline src/components/monaco-query-field/monaco-completion-provider/getDottedIdentifierBounds.ts:6]: Can we get an infinite loop here?

Comment (arturminchukov) [reply src/components/monaco-query-field/monaco-completion-provider/getDottedIdentifierBounds.ts:6]: No, `start` every iteration is decreased by 1, so the condition `start > 0` will not be true.
